### PR TITLE
remove netcoreapp2.1 from projects. clean up csproj

### PR DESCRIPTION
**Pull request by TimothyMothra**
Fix Issue #2251 .

## Changes
- removing `netcoreapp2.1` from projects
- cleaning up csproj

### Checklist
- [ ] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

File: BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Microsoft.ApplicationInsights.Tests.csproj [hunk 1]
```diff
@@ -2,8 +2,8 @@
   <Import Project="$(PropsRoot)\Test.props" />
   
   <PropertyGroup>
-    <TargetFrameworks>net452;net46;net461;netcoreapp2.1;netcoreapp3.1;net5.0</TargetFrameworks>
-    <TargetFrameworks Condition="$(OS) != 'Windows_NT'">netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net452;net46;net461;netcoreapp3.1;net5.0</TargetFrameworks>
+    <TargetFrameworks Condition="$(OS) != 'Windows_NT'">netcoreapp3.1</TargetFrameworks>
```
Comment (cijothomas) [inline BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Microsoft.ApplicationInsights.Tests.csproj:6]: we can add net5.0 in non-windows

Comment (TimothyMothra) [reply BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Microsoft.ApplicationInsights.Tests.csproj:6]: i want to address net5.0 as a separate PR.

File: BASE/src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj [hunk 2]
```diff
@@ -35,4 +35,4 @@
     <AdditionalFiles Include="$(PublicApiRoot)\$(AssemblyName).dll\$(TargetFramework)\PublicAPI.Unshipped.txt" />
   </ItemGroup>
   
-</Project>
\ No newline at end of file
+</Project>
```
Comment (cijothomas) [inline BASE/src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj:38]: nothing changed?

File: BASE/src/ServerTelemetryChannel/TelemetryChannel.csproj [hunk 3]
```diff
@@ -28,6 +28,10 @@
     <Reference Include="System.Net.Http" />
   </ItemGroup>
   
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
```
Comment (cijothomas) [inline BASE/src/ServerTelemetryChannel/TelemetryChannel.csproj:31]: nothing changed right?

**Review by cijothomas — APPROVED** [review]

File: BASE/src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj (hunk 2 above)
Comment (TimothyMothra) [reply BASE/src/Microsoft.ApplicationInsights/Microsoft.ApplicationInsights.csproj:38]: Github complains if a file is missing a new-line at the very end. That's the red circle on line 38.

**Pull request merged**